src/client/datascience/kernel-launcher/types.ts

```diff
@@ -28,6 +29,9 @@ export interface IKernelConnection {
 export interface IKernelProcess extends IDisposable {
     process: ChildProcess | undefined;
     connection: IKernelConnection | undefined;
+    ready: Promise<void>;
+    kernelSpec: IJupyterKernelSpec | undefined;
```

Comment:
Would prefer if this was a readonly property. 
Cuz I know we mutate these things a lot in existing code.
```suggestion
    readonly kernelSpec: Readonly<IJupyterKernelSpec> | undefined;
```

Reply:
If you agree Please could you also make `connection` a readonly property and return a `Readonly<IKernelConnection>` as well.

Reply:
Super good call. We need way more readonly in our code.